src/graph.jl

```diff
@@ -28,10 +28,10 @@ function DispatchGraph(
     input_nodes::Union{AbstractArray{S}, Base.AbstractSet{S}}=DispatchNode[],
 ) where {T<:DispatchNode, S<:DispatchNode}
     graph = DispatchGraph()
-    to_visit = Stack(DispatchNode)
+    to_visit = Stack{DispatchNode}()
 
     # this is an ObjectIdDict to avoid a hashing stack overflow when there are cycles
-    visited = ObjectIdDict()
+    visited = IdDict{Any, Any}()
```

Comment:
Is `IdDict` in Compat? I don't think it is, and if not, we'll have to do a `VERSION` conditional here.